skynet/modules/ttt/summaries/processor.py

```diff
@@ -1,48 +1,58 @@
-
 from langchain.chains.summarize import load_summarize_chain
-from langchain.llms import LlamaCpp
-from langchain.prompts import PromptTemplate
+from langchain.prompts import ChatPromptTemplate
 from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain_core.documents import Document
+from langchain_openai import ChatOpenAI
 
-from skynet.env import llama_n_batch, llama_n_gpu_layers, llama_path
+from skynet.env import openai_api_base_url
+from skynet.logs import get_logger
 
-from .prompts.action_items import action_items_template
-from .prompts.summary import summary_template
-from .v1.models import Job, JobType
+from .prompts.action_items import action_items_system_message
+from .prompts.summary import summary_system_message
+from .v1.models import JobType
 
-executor = None
 llm = None
+map_reduce_threshold = 12000
+log = get_logger(__name__)
 
 
 def initialize():
-    global executor, llm
-    executor = ThreadPoolExecutor(max_workers=1)
-
-    llm = LlamaCpp(
-        model_path=llama_path,
-        temperature=0.01,
-        max_tokens=4096,
-        n_ctx=4096,
-        n_gpu_layers=llama_n_gpu_layers,
-        n_batch=llama_n_batch,
+    global llm
+
+    llm = ChatOpenAI(
+        api_key='placeholder',  # use a placeholder value to bypass validation, and allow the custom base url to be used
+        base_url=openai_api_base_url,
+        temperature=0,
     )
 
 
-async def process(job: Job) -> str:
-    text = job.payload.text
+async def process(text: str, job_type: JobType, model: ChatOpenAI = None) -> str:
+    current_model = model or llm
+    chain = None
 
     if not text:
         return ""
 
-    loop = asyncio.get_running_loop()
-    text_splitter = RecursiveCharacterTextSplitter(chunk_size=12000, chunk_overlap=100)
-    docs = text_splitter.create_documents([text])
-    template = summary_template if job.type is JobType.SUMMARY else action_items_template
+    system_message = summary_system_message if job_type is JobType.SUMMARY else action_items_system_message
+    prompt = ChatPromptTemplate.from_messages([("system", system_message), ("user", "{text}")])
+
+    if len(text) < map_reduce_threshold:
+        chain = load_summarize_chain(current_model, chain_type="stuff", prompt=prompt)
+        docs = [Document(page_content=text)]
+    else:
+        text_splitter = RecursiveCharacterTextSplitter(chunk_size=map_reduce_threshold, chunk_overlap=100)
+        docs = text_splitter.create_documents([text])
+        chain = load_summarize_chain(current_model, chain_type="map_reduce", combine_prompt=prompt, map_prompt=prompt)
 
-    prompt = PromptTemplate(template=template, input_variables=["text"])
+    result = await chain.ainvoke(input={"input_documents": docs})
 
-    chain = load_summarize_chain(llm, chain_type="map_reduce", combine_prompt=prompt)
+    return result['output_text'].strip()
+
+
+async def process_open_ai(text: str, job_type: JobType, api_key: str) -> str:
+    llm = ChatOpenAI(
```

Comment:
Would it make sense to have an LRU cache with there, rather than (potentially) build the same object over and over?

Reply:
i dont think this is costly enough to need that